src/lib.rs

```diff
@@ -178,22 +139,18 @@ pub fn run(args: Args) {
         _ => unreachable!(),
     };
     let specified_supported_language = args.language.map(|language| language.get_language());
-    let query_or_failure_by_language: Mutex<HashMap<SupportedLanguageName, Option<Arc<Query>>>> =
-        Default::default();
+    let cached_queries: CachedQueries = Default::default();
     let capture_index = MaybeInitializedCaptureIndex::default();
-    let output_mode = get_output_mode(&args);
+    let output_mode = args.output_mode();
     let buffer_writer = BufferWriter::stdout(ColorChoice::Never);
 
-    get_project_file_walker(specified_supported_language.as_deref(), &args.use_paths())
-        .into_parallel_iterator()
+    get_project_file_parallel_iterator(specified_supported_language.as_deref(), &args.use_paths())
```

Comment:
As long as this was going in a separate module it started to seem strange to even expose it as a `WalkParallel` (and then explicitly call `.into_parallel_iterator()` on that) (since from this consumer's perspective it can be oblivious to the fact that it ever was a `WalkParallel`) so encapsulated that